conf/application.conf

```diff
@@ -39,7 +39,35 @@ play.filters.enabled += "filters.AllowListFilter"
 # Custom error handler
 play.http.errorHandler = "config.ErrorHandler"
 
-play.filters.headers.contentSecurityPolicy = "default-src 'self' 'unsafe-inline' localhost:12345 www.google-analytics.com www.googletagmanager.com tagmanager.google.com data:"
+play.filters.enabled += play.filters.csp.CSPFilter
+
+play.filters.csp {
+    nonce {
+      enabled = true
+      pattern = "%CSP_NONCE_PATTERN%"
+      header = true
+    }
+
+    directives {
+      base-uri = "'self'"
+      block-all-mixed-content = ""
+      child-src = "'none'"
+      connect-src = "'self' https://www.google-analytics.com localhost:12345 https://www.googletagmanager.com"
+      default-src = "'none'"
+
+      font-src = "'self'"
+      form-action = "'self'"
+
+      frame-ancestors = "'self'"
+      img-src = "'self' https://www.googletagmanager.com"
+
+      script-src = ${play.filters.csp.nonce.pattern} "'self' https://www.google-analytics.com https://www.googletagmanager.com 'sha256-+6WnXIl4mbFTCARd8N3COQmT3bJJmo32N8q8ZSQAIcU='"
+      style-src = ${play.filters.csp.nonce.pattern} "'self'"
+
+      report-to = "http://localhost:7950/register-an-estate/csp/report/report-to"
```

Comment:
@scullythornton1 Is this URL correct?

Reply:
sorry I will remove this for now, do not need it